File: macos-app/Tests/ThoughtInputTests/PendingCaptureStoreTests.swift
import XCTest
@testable import ThoughtInput

final class PendingCaptureStoreTests: XCTestCase {

    private var store: PendingCaptureStore!

    private func makePendingCapture(text: String, method: CapturePayload.CaptureMethod) -> PendingCapture {
        let payload = CapturePayload.create(text: text, method: method)
        let destination = Destination(
            name: "Test",
            isActive: true,
            type: .restNoAuth(RESTNoAuthConfig(endpointURL: "https://example.com"))
        )
        return PendingCapture(payload: payload, destinationID: destination.id, destinationSnapshot: destination)
    }

    override func setUp() {
        super.setUp()
        store = PendingCaptureStore()
    }

    override func tearDown() {
        let pending = store.loadAll()
        for capture in pending {
            store.remove(idempotencyKey: capture.payload.idempotencyKey)
        }
        super.tearDown()
    }

    func testSaveAndLoad() {
        let capture = makePendingCapture(text: "Test pending", method: .typed)
        store.save(capture)

        let loaded = store.loadAll()
        XCTAssertTrue(loaded.contains(where: { $0.payload.idempotencyKey == capture.payload.idempotencyKey }))
    }

    func testRemove() {
        let capture = makePendingCapture(text: "Remove me", method: .voice)
        store.save(capture)
        store.remove(idempotencyKey: capture.payload.idempotencyKey)

        let loaded = store.loadAll()
        XCTAssertFalse(loaded.contains(where: { $0.payload.idempotencyKey == capture.payload.idempotencyKey }))
    }

    func testPendingCount() {
        let initialCount = store.pendingCount
        let capture = makePendingCapture(text: "Count test", method: .typed)
        store.save(capture)

        XCTAssertEqual(store.pendingCount, initialCount + 1)

        store.remove(idempotencyKey: capture.payload.idempotencyKey)
        XCTAssertEqual(store.pendingCount, initialCount)
    }
}

File: windows-app/src/ThoughtInput.Tests/PendingCaptureStoreTests.cs
using System.Text.Json;
using FluentAssertions;
using ThoughtInput.Core;
using Xunit;

namespace ThoughtInput.Tests;

public class PendingCaptureStoreTests : IDisposable
{
    private readonly string _dir;

    public PendingCaptureStoreTests()
    {
        _dir = Path.Combine(Path.GetTempPath(), "ti-pending-" + Guid.NewGuid().ToString("N"));
    }

    public void Dispose()
    {
        try { Directory.Delete(_dir, recursive: true); } catch { }
    }

    private static PendingCapture MakePending(string text = "thought")
    {
        var payload = CapturePayload.Create(text, CaptureMethod.Typed);
        var dest = new Destination
        {
            Name = "n",
            Type = new RestNoAuthDestinationType(new RestNoAuthConfig("https://example.com")),
        };
        return new PendingCapture(payload, dest.Id, dest);
    }

    [Fact]
    public void Save_LoadAll_Round_Trips()
    {
        var store = new PendingCaptureStore(_dir);
        var p = MakePending();
        store.Save(p).Should().BeTrue();

        var loaded = store.LoadAll();
        loaded.Should().HaveCount(1);
        loaded[0].Payload.IdempotencyKey.Should().Be(p.Payload.IdempotencyKey);
        loaded[0].Payload.Text.Should().Be("thought");
    }

    [Fact]
    public void Remove_DeletesFile()
    {
        var store = new PendingCaptureStore(_dir);
        var p = MakePending();
        store.Save(p);
        store.PendingCount.Should().Be(1);

        store.Remove(p.Payload.IdempotencyKey);
        store.PendingCount.Should().Be(0);
    }

    [Fact]
    public void LoadAll_ReadsLegacyBareCapturePayload()
    {
        Directory.CreateDirectory(_dir);
        var legacyPayload = CapturePayload.Create("legacy", CaptureMethod.Typed);
        var bytes = JsonSerializer.SerializeToUtf8Bytes(legacyPayload, DestinationJson.DefaultOptions);
        File.WriteAllBytes(Path.Combine(_dir, $"{legacyPayload.IdempotencyKey}.json"), bytes);

        var store = new PendingCaptureStore(_dir);
        var loaded = store.LoadAll();

        loaded.Should().HaveCount(1);
        loaded[0].Payload.Text.Should().Be("legacy");
        loaded[0].DestinationSnapshot.Type.Should().BeOfType<RestNoAuthDestinationType>();
    }
}
